addons/account_tax_python/models/account_tax.py

```diff
@@ -29,33 +30,68 @@ class AccountTaxPython(models.Model):
             ":param product: product.product recordset singleton or None\n"
             ":param partner: res.partner recordset singleton or None")
 
-    def _compute_amount(self, base_amount, price_unit, quantity=1.0, product=None, partner=None, fixed_multiplicator=1):
-        self.ensure_one()
-        if product and product._name == 'product.template':
-            product = product.product_variant_id
-        if self.amount_type == 'code':
-            company = self.env.company
-            localdict = {'base_amount': base_amount, 'price_unit':price_unit, 'quantity': quantity, 'product':product, 'partner':partner, 'company': company}
+    def _prepare_dict_for_taxes_computation(self):
+        # EXTENDS 'account'
+        tax_values = super()._prepare_dict_for_taxes_computation()
+        tax_values.update({
+            'python_compute': self.python_compute,
+            'python_applicable': self.python_applicable,
```

Comment:
Why can't we do:
```suggestion
            'python_tax_id': self.id,
```
and do a browse for the safe_eval? I don't think it's used anywhere else

Reply:
@mart-e Since it only works python-side, you are right :+1: